src/Core/Dashboard/components/RefferalCode/RefferalCode.tsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import { Box, Button, Flex, Text } from '@chakra-ui/react';
+import { getReferralCode } from '../../../../shared/CrowdsourcerAPI';
+import copyToClipboard from '../../../../shared/utils/copyToClipboard';
+import * as Icons from '../../../Collections/iconography';
+
+export const RefferalCode = (): React.ReactElement => {
+  const [referralCode, setReferralCode] = React.useState('');
+  const ref = React.useRef<HTMLParagraphElement>();
+
+  React.useEffect(() => {
+    getReferralCode().then(setReferralCode);
+  }, []);
+
+  const onCopyReferralCode = () => {
+    const range = window.document.createRange();
+    range.selectNode(ref.current);
+    window.getSelection().removeAllRanges();
+    window.getSelection().addRange(range);
+
+    copyToClipboard({
+      copyText: ref.current.textContent,
+    });
+  };
```

Comment:
could you do something like this?:
```suggestion
  const onCopyReferralCode = () => {
    if (referralCode) {
      copyToClipboard({
        copyText: referralCode
      });
    }
  };
```

Reply:
refactored, Thanks!